### PR TITLE
Base Theming

### PR DESCRIPTION
**Pull request by carlos-kelly**
This PR adds base theming via Styled System and Styled Components plus a handful of Text components like Header and List.

Each component is a styled component with themed defaults. Meaning the user can either customize it with styled system props `<Text fontSize />` or extend it using Styled Components directly.

Styled theme props can either take the name of a theme label or a raw value
`<Text fontSize="big" />` or `<Text fontSize="24px" />`. This PR has a default theme based on Formidable colors.

Screen Shot 2019-09-04 at 12 34 40 PM

### CODE REVIEW

File: src/components/deck.js [hunk 1]
```diff
@@ -111,7 +114,15 @@ const Deck = ({ children, loop, keyboardControls, ...rest }) => {
...
     document.body.style.margin = '0';
-  }, []);
+    document.body.style.background =
+      themeContext.colors[rest.backgroundColor] ||
+      rest.backgroundColor ||
+      themeContext.colors.tertiary;
+    document.body.style.color =
+      themeContext.colors[rest.textColor] ||
+      rest.textColor ||
+      themeContext.colors.primary;
```
Comment (carlos-kelly) [inline src/components/deck.js]: We want to set the body color to the theme background and text color to theme text color. We can source the theme from either
1. The theme label
2. A raw value
3. The default theme value

File: src/components/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  Slide,
+  SlideElementWrapper,
+  CodePane,
+  OrderedList,
+  Quote,
+  Heading,
+  ListItem,
+  UnorderedList,
+  Text
+};
```
Comment (carlos-kelly) [inline src/components/index.js]: This unifies all of our exports into a single point.

File: src/theme/default-theme.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    header: '48px',
+    subHeader: '32px',
+    text: '18px'
+  },
+  space: {
+    headerMargin: '16px',
+    textMargin: '16px',
+    listMargin: '16px 0'
+  }
+};
```
Comment (carlos-kelly) [inline src/theme/default-theme.js]: These are some defaults for our theme's space, colors, and fonts. https://styled-system.com/getting-started

**Comment by rgerstenberger** [comment]
Does theming allow for adding a logo (say Formidable's ;)) somewhere on all slides (say bottom right)?

File: src/theme/default-theme.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+export default {
+  colors: {
+    primary: '#ebe5da',
+    secondary: '#f04d21',
+    tertiary: '#1e2852'
+  },
+  fonts: {
+    header: 'Helvetica Neue',
```
Comment (rgerstenberger) [inline src/theme/default-theme.js]: Looks like this should include fallback fonts?

File: src/components/typography.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+Text.defaultProps = {
+  color: 'primary',
+  fontFamily: 'text',
+  fontSize: 'text',
+  textAlign: 'left',
+  padding: 0,
+  margin: 'textMargin'
+};
+
+const Heading = styled(Text)({});
```
Comment (SunburtReynolds) [inline src/components/typography.js]: Do you have to include `({})`?

File: src/theme/default-theme.js (hunk 3 above)
Comment (SunburtReynolds) [reply src/theme/default-theme.js]: Do you think we should make our theme public/available to the public?

File: src/components/typography.js (hunk 5 above)
Comment (carlos-kelly) [reply src/components/typography.js]: I have to do something either `({})` or back-ticks. IMO, a blank object looks less strange.

File: src/theme/default-theme.js (hunk 3 above)
Comment (carlos-kelly) [reply src/theme/default-theme.js]: Yes! I plan on creating a helper that lets you use or extend it, which really is Object.assign. But still a helper.

**Pull request merged**